Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -296,6 +297,143 @@ namespace Babylon
+            else
+            {
+                m_handle = bgfx::DynamicVertexBufferHandle{bgfx::kInvalidHandle};
+            }
+        }
+
+        ~VertexBufferData()
+        {
+            DoForHandleTypes([](auto handle) {
+                if (handle.idx != bgfx::kInvalidHandle)
+                {
+                    bgfx::destroy(handle);
+                } }, [](auto handle) {
+                if (handle.idx != bgfx::kInvalidHandle)
+                {
+                    bgfx::destroy(handle);
+                } });
+        }
+
+        void EnsureFinalized(Napi::Env env, const bgfx::VertexLayout& layout)
+        {
+            DoForHandleTypes([&layout, this](auto handle) {
+                if (handle.idx != bgfx::kInvalidHandle)
+                {
+                    return;
+                }
+
+                const bgfx::Memory* memory = bgfx::makeRef(m_bytes.data(), m_bytes.size(), [](void*, void* userData)
+                {
+                    auto* bytes = reinterpret_cast<std::vector<uint8_t>*>(userData);
+                    bytes->clear();
+                }, &m_bytes);
+
+                m_handle = bgfx::createVertexBuffer(memory, layout); }, [&layout, this](auto handle) {
+                if (handle.idx != bgfx::kInvalidHandle)
+                {
+                    return;
+                }
+
+                const bgfx::Memory* memory = bgfx::makeRef(m_bytes.data(), m_bytes.size(), [](void*, void* userData)
+                {
+                    auto* bytes = reinterpret_cast<std::vector<uint8_t>*>(userData);
+                    bytes->clear();
+                }, &m_bytes);
+
+                m_handle = bgfx::createDynamicVertexBuffer(memory, layout); });
+        }
+
+        void Update(const Napi::Uint8Array& bytes, uint32_t offset, uint32_t byteLength)
+        {
+            DoForHandleTypes([](auto) { throw std::runtime_error("Cannot update non-dynamic vertex buffer."); }, [&bytes, offset, byteLength, this](auto handle) {
+                if (handle.idx == bgfx::kInvalidHandle)
+                {
+                    // Buffer hasn't been finalized yet, all that's necessary is to swap out the bytes.
+                    m_bytes = {bytes.Data() + offset, bytes.Data() + offset + byteLength};
+                }
+                else
+                {
+                    // Buffer was already created, do a real update operation.
+                    const bgfx::Memory* memory = bgfx::copy(bytes.Data() + offset, byteLength);
+                    bgfx::update(handle, 0, memory);
+                } });
+        }
+
+        void SetAsBgfxVertexBuffer(uint8_t index, uint32_t startVertex, bgfx::VertexLayoutHandle layout) const
+        {
+            DoForHandleTypes([index, startVertex, layout](auto handle) { bgfx::setVertexBuffer(index, handle, startVertex, UINT32_MAX, layout); }, [index, startVertex, layout](auto handle) { bgfx::setVertexBuffer(index, handle, startVertex, UINT32_MAX, layout); });
```

Comment:
I'm not even going to pretend to understand why clang format thought this was a good idea.

It looks like there's at least some support for less appalling [Allman-style](https://en.wikipedia.org/wiki/Indentation_style#Allman_style) lambda formatting in clang format 11. I think VS code can do clang format 11, and there's an add-on to make VS 2019 do it. Should we explore that further?

Reply:
This went away when I had to rephrase the code for clang.